spec/compilers/component_instance_access

```diff
@@ -14,8 +14,11 @@ component Instance {
 }
 
 component Main {
-  fun handleClick : Maybe(String) {
-    instance&.name
```

Comment:
This seemed like a handly, QoL language feature, why did you decide to remove it?

Reply:
1. It was not documented
2. I wanted to simply the language
3. There might be better ways to allow this behavior

Reply:
I don't see these as a reasons for the removal.

1. We could document it
2. I agree it's important, but OTOH this construct simplifies the language for the developer
3. Every language I know that deals with nil/null values ends up with a similar construct, what alternatives you see here?

Reply:
> I agree it's important, but OTOH this construct simplifies the language for the developer

In simple occurrences like in this example sure, but if there are multiple `Maybe`s, it compounds, and makes it harder to understand what is going on.

 > Every language I know that deals with nil/null values ends up with a similar construct, what alternatives you see here?
 
I'm not sure, what Crystal does with `object.try(&.call.chain)` is the best I've seen so far, how can that be applied here I don't know. We already have the `or` keyword to fall back to a default value maybe that's sufficient for now.